### ISSUE TITLE
Pedigree field causes error

### ISSUE THREAD

**Issue by xiamaz** [bihealth/mehari#176]
Currently noodles-vcf `0.34` does not support pedigree entries in VCFv4.2 format. Issue in noodles has been created here: https://github.com/zaeleus/noodles/issues/201

Migrating to a newer noodles-vcf version causes non-straightforward migration issues, as multiple field types have been removed from upstream noodes, including the `META` header field, which is being used in our code.

Currently the following patch fixes the issue on the `0.34` version of noodles:

```
...
                let v = match value {
                    parser::Value::String(s) => Value::from(s),
                    parser::Value::Struct(mut fields) => {
                        let id = remove_field_or_first(&mut fields, ID).ok_or_else(|| { // <-- replace this line with the new function
                            ParseError::InvalidOther(
                                k.clone(),
                                map::other::ParseError::MissingField(map::other::tag::ID),
                            )
                        })?;

                        let map = Map::<Other>::try_from(fields)
                            .map_err(|e| ParseError::InvalidOther(k.clone(), e))?;

                        Value::from((id, map))
                    }
                };

                Ok(Self::Other(k, v))
.,..
fn remove_field_or_first(fields: &mut Vec<(String, String)>, key: &str) -> Option<String> {
    let i = fields.iter().position(|(k, _)| k == key).unwrap_or(0);
    let (_, value) = fields.remove(i);
    Some(value)
}
```

**Comment by xiamaz** [comment]
This issue has been fixed in `noodles-vcf` upstream, which is currently not released. Given updating to the newest release is not trivial, it might be a good idea starting migration to a newer noodles-version now.

https://github.com/zaeleus/noodles/issues/201

**Comment by xiamaz** [comment]
Built mehari using upstream noodles, confirmed this error will be fixed. Bump Version once the patches are released in `noodles`.

**Comment by xiamaz** [comment]
Solved with ec00074
